packages/nameguard-sdk/src/index.ts

```diff
@@ -376,15 +376,15 @@ export interface ConsolidatedNameGuardReport extends ConsolidatedReport {
 /**
  * NameGuard report that contains the full results of all `checks` on all `labels` in a name.
  */
-export interface NameGuardReport extends ConsolidatedNameGuardReport {
+export interface AbstractNameGuardReport extends ConsolidatedNameGuardReport {
```

Comment:
Suggest we add a new field to `AbstractNameGuardReport`:

`inspected: boolean`

Then, in `InspectedNameGuardReport` define this field as:

`inspected: true`

And in `UninspectedNameGuardReport` define this field as:

`inspected: false`

Then please update the code that returns these to set the value for this field appropriately.

The benefit of this extra effort is it helps to make all the code that uses this more self documenting.

For example, a function will take as input a `NameGuardReport`. The type system will know this is either a InspectedNameGuardReport | UninspectedNameGuardReport.

Then, you can write code like the following:

```
if (report.inspected) {
  // in this code block the type system knows that `report` is an `InspectedNameGuardReport`.
} else {
  // in this code block the type system knows that `report` is an `UninspectedNameGuardReport`.
}
```

Reply:
@lightwalker-eth "Then please update the code that returns these to set the value for this field appropriately." - do you mean Python or TS code?

Reply:
@djstrong Ok cool. The goal of that statement is to ensure the value is set in Python code and then ultimately flows correctly into the TS code.